src/components/RightSidebar/PinnedMessages/PinnedMessageItem.vue

```diff
@@ -0,0 +1,126 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+<script setup lang="ts">
+import type { ChatMessage } from '../../../types/index.ts'
+
+import { t } from '@nextcloud/l10n'
+import { computed } from 'vue'
+import { useRoute } from 'vue-router'
+import { useStore } from 'vuex'
+import NcActionButton from '@nextcloud/vue/components/NcActionButton'
+import NcListItem from '@nextcloud/vue/components/NcListItem'
+import IconPinOff from 'vue-material-design-icons/PinOffOutline.vue'
+import IconPin from 'vue-material-design-icons/PinOutline.vue'
+import { useGetToken } from '../../../composables/useGetToken.ts'
+import { AVATAR } from '../../../constants.ts'
+import { EventBus } from '../../../services/EventBus.ts'
+import { useSharedItemsStore } from '../../../stores/sharedItems.ts'
+import { formatDateTime } from '../../../utils/formattedTime.ts'
+import { parseToSimpleMessage } from '../../../utils/textParse.ts'
+
+type PinnedChatMessage = ChatMessage & {
+	metaData: {
+		pinnedActorId: string
+		pinnedActorDisplayName: string
+		pinnedUntil: number
+		pinnedAt: number
+	}
+}
```

Comment:
```suggestion
type PinnedChatMessage = Omit<ChatMessage, 'metaData'> & Required<Pick<ChatMessage, 'metaData'>>
```

Should be in types/index.ts maybe?